### PR TITLE
get/set network connections

### PR DESCRIPTION
**Pull request by scottdixon**
@moizjv, @imurchie, @jlipps

### CODE REVIEW

File: lib/commands/network.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  type >>= 1;
+  let wifi = type % 2;
+  type >>= 1;
+  let data = type % 2;
+
+  await this.handleADBDisconnectAndPerformAction('setAirplaneMode', airplaneMode);
+  await this.handleADBDisconnectAndPerformAction('broadcastAirplaneMode', airplaneMode);
+
+  if (!airplaneMode){
+    await this.handleADBDisconnectAndPerformAction('setWifiAndData', {wifi, data});
```
Comment (scottdixon) [inline lib/commands/network.js]: Should {wifi, data} be broken into two lines?

Comment (imurchie) [inline lib/commands/network.js]: I would say not.

But there ought to be a space before the `{`.

File: lib/commands/network.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  }
+
+  return await this.getNetworkConnection();
+};
+
+helpers.handleADBDisconnectAndPerformAction = async function(action, params){
```
Comment (imurchie) [inline lib/commands/network.js]: Space after `function`, and another before `{`

Comment (jlipps) [inline lib/commands/network.js]: spacing should be: `async function (action, params) {`

File: lib/commands/network.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  this.ignoreUnexpectedShutdown = true;
+  await this.adb[action](params);
+  await this.adb.restart();
+  await this.bootstrap.start(this.opts.appPackage, this.opts.disableAndroidWatchers);
```
Comment (jlipps) [inline lib/commands/network.js]: the fact that the uiautomator bootstrap shuts down makes me think that this function is really `handleUiAutomatorDisconnectAndPerformAction`. The long name also makes me wonder if we should be architecting this differently. What if it was:

```js
// how we call it
await this.wrapBootstrapDisconnect(async () => {
  await this.adb.setAirplaneMode(airplaneMode);
});

// how it's defined
helpers.wrapBootstrapDisconnect = async function (wrapped) {
  this.ignoreUnexpectedShutdown = true;
  await wrapped();
  await this.adb.restart();
  await this.bootstrap.start(this.opts.appPackage, this.opts.disableAndroidWatchers);
};
```

This seems a lot cleaner and has the advantage of having clearer names :-)

File: lib/commands/network.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+  type >>= 1;
+  let data = type % 2;
+
+  await this.wrapBootstrapDisconnect(async () => {
+    await this.adb.setAirplaneMode(airplaneMode);
+  });
+  await this.wrapBootstrapDisconnect(async () => {
+    await this.adb.broadcastAirplaneMode(airplaneMode);
+  });
+  if (!airplaneMode){
```
Comment (jlipps) [inline lib/commands/network.js]: I think @imurchie meant that there should be a space before this brace, so it looks like:

```js
if (!airplaneMode) {
```

File: lib/commands/network.js [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+    await this.wrapBootstrapDisconnect(async () => {
+      await this.adb.setWifiAndData({ wifi, data });
```
Comment (jlipps) [inline lib/commands/network.js]: and then this one should be as you had it:

```js
{wifi, data}
```

the general rule is that JS objects don't have spaces around braces, but functions/control structures/imports/exports do (so as to distinguish them among other things)

**Comment by jlipps** [comment]
also, looks like some rebasing is required. i pushed some stuff so might have gotten in your way. let me know if it's not obvious how to resolve the conflicts

**Pull request merged**